### PR TITLE
[Scala3] Allow empty template with `:` + end marker for object/class/trait

### LINKED ISSUE #2387: Scala 3 empty object with `end` token error
In Scala 3 dialect (under Scalafmt) we get parse error when an object has no body, but has end token.
```scala
object Foo:
end Foo
```

### PR DESCRIPTION
**Pull request by dos65**
Fixes #2387

### CODE REVIEW

File: scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala [hunk 1]
```diff
@@ -907,6 +907,7 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
   def acceptStatSep(): Unit = token match {
     case LF() | LFLF() => next()
     case _ if in.observeOutdented() =>
+    case t if t.is[EndMarkerIntro] =>
```
Comment (tgodzik) [inline scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala:910]: Is this needed? Outdent should be always before end marker and from what I see we don't need it below.

**Review by tgodzik — CHANGES_REQUESTED** [review]

File: scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala (hunk 1 above)
Comment (dos65) [reply scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala:910]: For empty template body, there is no indent/outdent
```scala
object Foo: // <- here at colon  we call `observeIndented` while the next line doesn't have any indentation
end Foo
```

Comment (tgodzik) [reply scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala:910]: Generally, it should end on the LF after `:`, but this doesn't seem to happen and we are skipping the LF. That would be the underlying issue here. However, not sure how to fix it properly, so it's ok for now.

**Review by tgodzik — APPROVED** [review]
LGTM

File: scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala (hunk 1 above)
Comment (kitbellew) [reply scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala:910]: and would this work:
```
object Foo:
// comment
end Foo
```

Comment (dos65) [reply scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala:910]: @kitbellew yep. Parser ignores comments in the same way as whitespaces

**Pull request merged**